hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/fsdataset/impl/FsDatasetImpl.java

```diff
@@ -2846,6 +2847,23 @@ public void addBlockPool(String bpid, Configuration conf)
     if (volumeExceptions.hasExceptions()) {
       throw volumeExceptions;
     }
+
+    if(!blockPoolId.isEmpty()) {
+      bpid = blockPoolId;
+    }
+    ArrayList<String> pmemVolumes = PmemVolumeManager.getInstance().getVolumes();
```

Comment:
Line 2854 -2859 should not be here. This logic should be executed inside PmemVolumeManager.